### ISSUE TITLE
Some questions about the calculationof FCC solid solution phase

### ISSUE THREAD

**Issue by duxiaoxian** [PhasesResearchLab/ESPEI-datasets#14]
For Ag-Cu binary phase diagram, there are only liquid phase and FCC solid solution phase. For Ag-Cu FCC solid solution phase(FCC_A1), "sublattice_model": [["AG", "CU"], ["VA"], "sublattice_site_ratios": [1,1].
In this case, how to calculate the enthalpy change of FCC solid solution at different temperatures by phonopy? If we use the primitive cell to calculate the enthalpy with temperature, such as Ag3Cu, "sublattice_occupancies" is empty, because this is stoichiometric phase, not solid solution phase, so I am very confused.

**Comment by bocklund** [comment]
In DFT our group commonly uses dilute substitution (e.g. 1 Cu atom substituted onto a Ag site in a 64 atom Ag supercell) or special quasirandom structures [1] for more concentrated mixing. 

You can see an example of how these are combined for CALPHAD modeling by looking at this paper by Xuan Liu [2]. Ozoliņš and others [3] calculated some compounds and SQS for Ag-Cu and a couple other related systems, which you might find insightful for how these types of calculations are compare to each other.

[1] https://doi.org/10.1103/PhysRevLett.65.353
[2] https://doi.org/10.1016/j.calphad.2013.02.006
[3] https://doi.org/10.1103/PhysRevB.57.6427

**Comment by duxiaoxian** [comment]
If 1 Cu atom substituted onto a Ag site in a 64 atom Ag supercell,  then "sublattice_occupancies": [ [0.015, 0.985], 1] , and then use this structure for phonon calculation. so that the thermodynamic properties of solid solution phase can be obtained. Am I right in understanding that?

**Comment by bocklund** [comment]
The occupancies should be [[0.985, 0.015], 1] (the elements must be in alphabetic order within a sublattice), but in principle your understanding is correct. You should be aware that the symmetry of the crystal will be lost, so phonon calculations may be expensive. For a computationally cheaper (but more approximate) calculation, you can use the Debye model, which is also briefly described in reference 2 in my response above.

**Comment by duxiaoxian** [comment]
Thank you very much for providing me with a new idea. I have not known debye model before. I found the relevant software on Google is gibbs2 or 1. May I ask what software you use for debye model calculation？
Another question is, I use the primordial cell containing four Ag atoms to replace one, two and three Cu atoms for phonon calculation, and write Ag3Cu, AgCu, AgCu3. Can this data be used as solid solution input, that is, for AgCu3, "sublattice_occupancies": [[0.25, 0.75], 1]?
I think it's strange. I don't think it's a solid solution.

**Comment by duxiaoxian** [comment]
When I used the SQS model from https://www.phases.psu.edu/?page_id=77 for fcc solid solution phase of Ag-Cu to calculating the static total energy by DFT, errors always occur because ions are too close or superimposed. So I always lack the data of thermodynamic properties of solid solutions. 😔

**Comment by bocklund** [comment]
The Debye model is fairly simple and implanted many places. I haven’t used the gibbs program, so I can’t comment on that. One of the easier and more flexible tools to get up and running is in `pymatgen` (See the [pymatgen Debye docs](http://pymatgen.org/pymatgen.analysis.quasiharmonic.html)). A drawback of their implementation is that the thermal electronic contributions to the free energy are not included. [dfttk](https://github.com/phasesresearchlab/dfttk) is a package that we have developed that, among other things, builds on top of pymatgen to include the thermal electronic part, but the package is still in development. 

You are correct that substituting Cu into the conventional cell creates compounds and not solid solutions, this is why the SQS were created :) - an alternative to predict the solution behavior using many compounds is the cluster expansion method.

For the SQS on the PRL website, the volume of that structure is probably too small for the atoms - you can just increase the volume by changing the scaling factor in the second row of the POSCAR file until the volume of the structure is large enough for the calculations to work. Usually a good rule of thumb for a guess of the volume is to make it so the structure has density equal to the weighted average density of fcc Ag and fcc Cu. Because of the atom collision issue, it’s usually safer to start with a cell that is a little larger, rather than a little smaller.

**Comment by duxiaoxian** [comment]
“Usually a good rule of thumb for a guess of the volume is to make it so the structure has density equal to the weighted average density of fcc Ag and fcc Cu. ” That is to say when expanding the volume, we should pay attention to the concentration ratio if the weighted average density is equal to the density of the new structure. Because ρ ̅=ρ_Ag X_Ag+ρ_Cu X_Cu=m_Ag/V+m_Cu/V，conversion formula we can obtain V_Ag/V_Cu =X_Cu/X_Ag =1.44.  Do I understand this correctly?

**Comment by bocklund** [comment]
I agree with the expression

ρ ̅=ρ_Ag X_Ag+ρ_Cu X_Cu

which will give you a target density of 

ρ ̅=10.49 g/cm^3 * 0.75 + 8.96 g/cm^3 * 0.25 = 10.1075 g/cm^3

This is your target density. You need to set the volume of your structure to match that target density. If you have pymatgen installed, this is pretty easy:

```python
from pymatgen import Structure
s = Structure.from_file('mysqs', fmt='POSCAR')

target_density = 10.1075
cur_volume = s.volume
cur_density = s.density
s.scale_lattice(s.volume * float((cur_density/target_density)))
print(s.density)  # should be the target

s.to(filename='scaled.sqs.POSCAR',  fmt='POSCAR')
```

**Comment by duxiaoxian** [comment]
Now I get it! Thank you for your patience. Your advice to me is very valuable.

**Comment by bocklund** [comment]
I'm happy to help!

**Comment by duxiaoxian** [comment]
According to your advice,  the volume of that structure was increased when using the SQS on the PRL website, such as the scaling factor is 2.  Is the total energy of the original SQS structure twice that of the changed structure? That is, E(SQS-origin)=E(SQS-changed)*2?

**Comment by duxiaoxian** [comment]
1. And if I want to calculate the phase diagram of the ternary system, for example, Ag-Cu-Co, What I should do for preparing input data and how to fit the ternary phase diagram by ESPEI.？
2. I have focused on calculating the thermodynamic data of the Ag-Cu binary system, should I do that on Cu-Co, Ag-Cu or just calculating the Ag-Cu-Co ternary system?  
To be honest, my main purpose is to calculate ternary phase diagrams😔.

**Comment by duxiaoxian** [comment]
I don't quite understand the volume-energy problem. If we expand the volume by two times, even according to the method of Ni and Ni3Al in the literature, we should calculate the volume by 1.5-2.5 times, and get the lowest energy after two times the volume, not the original SQS model energy that we want.

**Comment by bocklund** [comment]
> we should calculate the volume by 1.5-2.5 times, and get the lowest energy after two times the volume, not the original SQS model energy that we want.

This is correct. More generally, any structure you calculate should have all the degrees of freedom (cell volume, lattice parameters, and ionic positions) relaxed in a way that minimizes the forces on the atoms and the stress on the cell while preserving the symmetry of the structure.

**Comment by duxiaoxian** [comment]
I'm confused. What do you mean is that when I use the SQS model of PSU website to calculate formation energy, I need to set ISIF = 3?
In addition, the ionic distances of the original SQS structure are too close or the superposition results in errors in calculation, so I need to expand the volume by changing the scale factor of the second line of poscar. But in the end, we get the structural energy after enlarging the volume, which is not the original structural energy. 
Now I want the original structural energy if we calculate the enthalpy of mixing.
I don't understand the relationship between the energy calculated using the expanded SQS model and the original SQS model. I think these two should be two kinds of structures, and there should be some functional relationship between them.

**Comment by bocklund** [comment]
The “original”  energy of the SQS has no meaning. Since the energy would not be at a local minimum, the original energy represents the atoms trapped in a high energy state which is not useful to model. The only functional relationship between the energy and volume is the Lennard-Jones type curve, which the empirical equation of states discussed by Shang et al. 2010 (cited above) attempt to model. 

In principle you would relax the SQS with ISIF=3 to reach a meaningful local minimum in energy, but SQS in particular can sometimes be challenging to relax in a way that retains the symmetry of the parent structure, so we often relax the structure first by volume only (ISIF=7) until converged, then volume and shape (ISIF=6), making sure the shape change is not too large, then shape, volume and ions (ISIF=3), making sure the shape and/or ionic positions changes are not too large to break the symmetry.

**Comment by duxiaoxian** [comment]
Oh okay, now I get it! Thank you again for your patience to answer. Originally, my previous understanding has been wrong. You helped me clear my mind and let me be clear-minded. ❤🌹
